File: app/src/main/java/co/smartreceipts/android/model/impl/columns/categories/CategoryNameColumn.java
package co.smartreceipts.android.model.impl.columns.categories;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import java.util.List;

import co.smartreceipts.android.model.impl.columns.AbstractColumnImpl;
import co.smartreceipts.android.persistence.database.controllers.grouping.results.SumCategoryGroupingResult;
import co.smartreceipts.android.sync.model.SyncState;

public class CategoryNameColumn extends AbstractColumnImpl<SumCategoryGroupingResult> {

    public CategoryNameColumn(int id, @NonNull String name, @NonNull SyncState syncState) {
        super(id, name, syncState);
    }

    @NonNull
    @Override
    public String getValue(@NonNull SumCategoryGroupingResult sumCategoryGroupingResult) {
        return sumCategoryGroupingResult.getCategory().getName() + " [" + sumCategoryGroupingResult.getReceiptsCount() + "]";
    }

    @NonNull
    @Override
    public String getFooter(@NonNull List<SumCategoryGroupingResult> rows) {
        if (rows.isEmpty()) {
            return "";
        } else {
            return getValue(rows.get(0));
        }
    }
}

File: app/src/main/java/co/smartreceipts/android/model/impl/columns/categories/CategoryNameColumn.kt
package co.smartreceipts.android.model.impl.columns.categories

import co.smartreceipts.android.model.impl.columns.AbstractColumnImpl
import co.smartreceipts.android.persistence.database.controllers.grouping.results.SumCategoryGroupingResult
import co.smartreceipts.core.sync.model.SyncState

class CategoryNameColumn(id: Int, syncState: SyncState) :
    AbstractColumnImpl<SumCategoryGroupingResult>(
        id,
        CategoryColumnDefinitions.ActualDefinition.NAME,
        syncState
    ) {

    override fun getValue(rowItem: SumCategoryGroupingResult): String =
        rowItem.category.name + " [" + rowItem.receiptsCount + "]"
}
